SimTKcommon/Geometry/src/DecorativeGeometryRep.h

```diff
@@ -587,7 +595,7 @@ void implementGeometry(DecorativeGeometryImplementation& geometry) const overrid
 SimTK_DOWNCAST(DecorativeMeshFileRep, DecorativeGeometryRep);
 private:
 std::string meshFile;
-
+std::shared_ptr<PolygonalMesh> pmesh;
```

Comment:
> It is true that PolygonalMesh works as is but it doesn't or I couldn't find a method to tell if it has been populated/initialized. If you believe a separate flag (isInitialized) is cleaner I'll add that and remove the pointer.

You could use `PolygonalMesh::getNumVertices()` and assume that if the count is zero then it has not been initialized. That would only be wrong if the mesh file actually contained zero vertices which I suppose it could but that is pathological. In that case it would get reopened repeatedly, but I think that's fine as long as there is a comment pointing it out. A separate flag would be a little tidier.  

Using a pointer here is really just an awkward way of having a separate flag. Pointer==null is being used as the "uninitialized" flag.